### PR TITLE
[Python] Fix datetime with tzinfo converting to naive TIMESTAMP

### LINKED ISSUE #6466: [Python] datetime -> [duckdb] timestamptz conversion error
### What happens?

Incorrect timezone conversion.

### To Reproduce

```
from datetime import datetime, timezone
import duckdb

dt = datetime.now(timezone.utc).replace(microsecond=0)
print(dt)

q = "CREATE TABLE IF NOT EXISTS test (uid INT, dt TIMESTAMPTZ)"
duckdb.execute(q).commit()

q1 = "INSERT INTO test VALUES (?, ?)"
duckdb.execute(q1, (1, dt))     # BUG time zone
duckdb.execute(q1, (2, str(dt)))
duckdb.commit()

duckdb.sql("SELECT * FROM test").show()
```
```
2023-02-25 21:02:54+00:00
┌───────┬──────────────────────────┐
│  uid  │            dt            │
│ int32 │ timestamp with time zone │
├───────┼──────────────────────────┤
│     1 │ 2023-02-25 21:02:54+05   │
│     2 │ 2023-02-26 02:02:54+05   │
└───────┴──────────────────────────┘
```

### OS:

Linux x64, Windows x64

### DuckDB Version:

0.7.0

### DuckDB Client:

Python

### Full Name:

Leo

### Affiliation:

private

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

### PR DESCRIPTION
**Pull request by Tishj**
This PR fixes #6466 

When converting the `datetime.datetime` object, we already stripped the timezone, but didn't save the value as being of the type TIMESTAMPTZ

### CODE REVIEW

File: tools/pythonpkg/tests/fast/pandas/test_datetime_time.py [hunk 1]
```diff
@@ -69,3 +69,19 @@ def test_pandas_datetime_overflow(self):
 
         with pytest.raises(duckdb.ConversionException):
             res = duckdb_con.execute("select * from test").df()
+
+    def test_timezone_datetime(self):
+        module = pytest.importorskip("datetime")
```
Comment (Mause) [inline tools/pythonpkg/tests/fast/pandas/test_datetime_time.py:74]: `datetime` is built into Python, don't need to use importorskip for it

**Comment by davetapley** [comment]
@Tishj I just got around to testing this on 0.8.0 and it doesn't appear to have resolved the issues I raised on this comment: https://github.com/duckdb/duckdb/issues/6466#issuecomment-1489413242, specifically the implicit cast to UTC but still returning a naive Python `datetime`.


Just wanted to confirm before I open a new issue.

**Comment by Mytherin** [comment]
> @Tishj I just got around to testing this on 0.8.0 and it doesn't appear to have resolved the issues I raised on this comment: [#6466 (comment)](https://github.com/duckdb/duckdb/issues/6466#issuecomment-1489413242), specifically the implicit cast to UTC but still returning a naive Python `datetime`.
> 
> Just wanted to confirm before I open a new issue.

Feel free to open a new issue, I think that hasn't been implemented yet and it might get lost otherwise

**Comment by davetapley** [comment]
I don't see anything specific in [the 0.9.0 release notes](https://github.com/duckdb/duckdb/releases/tag/v0.9.0), but as a heads up: I just went from 0.8.0 to 0.9.0 and I'm now seeing this fixed.

**Pull request merged**